# **Classification of Claims in TikTok Videos**
**Third Stage - Hyopthesis Testing**

Situation: The current project is reaching its midpoint; a project proposal, Python coding work, and exploratory data analysis have all been completed.

Our next task is to determine and conduct the necessary hypothesis tests and statistical analysis for the TikTok classification project.


# **Data exploration and hypothesis testing**

In this activity, we will explore the data and conduct hypothesis testing.
<br/>

**The purpose** of this stage is to prepare, create, and analyze hypothesis tests.

**The goal** is to apply descriptive and inferential statistics, probability distributions, and hypothesis testing in Python.
<br/>

*This stage has three parts:*

**Part 1:** Imports and data loading. <br>
**Part 2:** Conducting hypothesis testing. <br>
**Part 3:** Communicating insights with stakeholders. <br>

# **PACE stages**

Throughout this notebook, we'll use references to the problem-solving framework PACE: Plan, Analyze, Construct, and Execute.

## **PACE: Plan**


The Question : Is there a relation between video views and author verification status? Is the relation statistically significant?

### **Part 1. Imports and Data Loading**

We'll Import packages and libraries needed to compute descriptive statistics and conduct a hypothesis test.

In [ ]:
# Importing packages for data manipulation

import pandas as pd
import numpy as np

# Importing packages for data visualization

import matplotlib.pyplot as plt
import seaborn as sns
# Importing packages for statistical analysis/hypothesis testing

from scipy import stats

In [ ]:
# Loading dataset into dataframe
data = pd.read_csv("tiktok_dataset.csv",index_col=0)

## **PACE: Analyze and Construct**

We use describtive statistics to identify center and dispertion of the data which help us detrmine main data characteristics and distribution as quickly as possible.

### **Part 2. Data exploration**

We'll use descriptive statistics to conduct Exploratory Data Analysis (EDA).



In [ ]:
# Displaying first few rows

data.head(10)

,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
#,,,,,,,,,,,
1,claim,7017666017,59,someone shared with me that drone deliveries a...,not verified,under review,343296.0,19425.0,241.0,1.0,0.0
2,claim,4014381136,32,someone shared with me that there are more mic...,not verified,active,140877.0,77355.0,19034.0,1161.0,684.0
3,claim,9859838091,31,someone shared with me that american industria...,not verified,active,902185.0,97690.0,2858.0,833.0,329.0
4,claim,1866847991,25,someone shared with me that the metro of st. p...,not verified,active,437506.0,239954.0,34812.0,1234.0,584.0
5,claim,7105231098,19,someone shared with me that the number of busi...,not verified,active,56167.0,34987.0,4110.0,547.0,152.0
6,claim,8972200955,35,someone shared with me that gross domestic pro...,not verified,under review,336647.0,175546.0,62303.0,4293.0,1857.0
7,claim,4958886992,16,someone shared with me that elvis presley has ...,not verified,active,750345.0,486192.0,193911.0,8616.0,5446.0
8,claim,2270982263,41,someone shared with me that the best selling s...,not verified,active,547532.0,1072.0,50.0,22.0,11.0
9,claim,5235769692,50,someone shared with me that about half of the ...,not verified,active,24819.0,10160.0,1050.0,53.0,27.0


In [ ]:
# Generating a table of descriptive statistics about the data

data.describe(include='all')

,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
count,19084,1.938200e+04,19382.000000,19084,19382,19382,19084.000000,19084.000000,19084.000000,19084.000000,19084.000000
unique,2,NaN,NaN,19012,2,3,NaN,NaN,NaN,NaN,NaN
top,claim,NaN,NaN,a friend read in the media a claim that badmi...,not verified,active,NaN,NaN,NaN,NaN,NaN
freq,9608,NaN,NaN,2,18142,15663,NaN,NaN,NaN,NaN,NaN
mean,NaN,5.627454e+09,32.421732,NaN,NaN,NaN,254708.558688,84304.636030,16735.248323,1049.429627,349.312146
std,NaN,2.536440e+09,16.229967,NaN,NaN,NaN,322893.280814,133420.546814,32036.174350,2004.299894,799.638865
min,NaN,1.234959e+09,5.000000,NaN,NaN,NaN,20.000000,0.000000,0.000000,0.000000,0.000000
25%,NaN,3.430417e+09,18.000000,NaN,NaN,NaN,4942.500000,810.750000,115.000000,7.000000,1.000000
50%,NaN,5.618664e+09,32.000000,NaN,NaN,NaN,9954.500000,3403.500000,717.000000,46.000000,9.000000
75%,NaN,7.843960e+09,47.000000,NaN,NaN,NaN,504327.000000,125020.000000,18222.000000,1156.250000,292.000000


In [ ]:
# Checking for missing values

data.isna().sum()

claim_status                0
video_id                    0
video_duration_sec          0
video_transcription_text    0
verified_status             0
author_ban_status           0
video_view_count            0
video_like_count            0
video_share_count           0
video_download_count        0
video_comment_count         0
dtype: int64

In [ ]:
# Droping rows with missing values


data = data.dropna().reset_index(drop=True)

In [ ]:
# Displaying first few rows after handling missing values


data.head()

,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
#,,,,,,,,,,,
1,claim,7017666017,59,someone shared with me that drone deliveries a...,not verified,under review,343296.0,19425.0,241.0,1.0,0.0
2,claim,4014381136,32,someone shared with me that there are more mic...,not verified,active,140877.0,77355.0,19034.0,1161.0,684.0
3,claim,9859838091,31,someone shared with me that american industria...,not verified,active,902185.0,97690.0,2858.0,833.0,329.0
4,claim,1866847991,25,someone shared with me that the metro of st. p...,not verified,active,437506.0,239954.0,34812.0,1234.0,584.0
5,claim,7105231098,19,someone shared with me that the number of busi...,not verified,active,56167.0,34987.0,4110.0,547.0,152.0


We are interested in the relationship between `verified_status` and `video_view_count`. One approach is to examine the mean value of `video_view_count` for each group of `verified_status` in the sample data.

In [ ]:
# Computing the mean `video_view_count` for each group in `verified_status`

data.groupby('verified_status')['video_view_count'].mean()

verified_status
not verified    265663.785339
verified         91439.164167
Name: video_view_count, dtype: float64

### **Part 3. Hypothesis testing**

**Null hypothesis:** There is no effect of aouthor verification status on number of video views. <br>
**Alternative hypothesis:** There is an effect of aouthor verification status on number of video views.



Our goal in this step is to conduct a two-sample t-test. The steps for conducting a hypothesis test are:


1.   Stating the null hypothesis and the alternative hypothesis
2.   Choosing a signficance level
3.   Finding the p-value
4.   Rejecting or failing to reject the null hypothesis



$H_0$ : There is no effect of author verification status on number of video views.<br>
$H_a$ : There is an effect of author verification status on number of video views.

We choosed 5% as the significance level and proceed with a two-sample t-test.

In [ ]:
# Conduct a two-sample t-test to compare means

verified = data[data['verified_status']=='verified']['video_view_count']
not_verified = data[data['verified_status']=='not verified']['video_view_count']
stats.ttest_ind(a=verified,b=not_verified, equal_var=False)

Ttest_indResult(statistic=-25.499441780633777, pvalue=2.6088823687177823e-120)

P-value = $2.6*10^{-120}$ which is much smaller than the significance level. Therfore, we `reject` the null hypothesis.

## **PACE: Execute**

## **Part 4: Communicating insights with stakeholders**

We can conclude that there is a strong effect of author verification status on number of video views which is by the way correlated with claim status. Where not verified authors tend to get much more views on average.